From 674330b0ccb816530ee6d31cea0f752e334f15d7 Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Tue, 19 Jul 2016 22:38:04 +0200
Subject: identity: Fix ensure_credentials_for_accounts() leak

The list returned by goa_client_get_accounts() is (transfer full).
ensure_credentials_for_accounts() was not freeing it.
---
 src/goaidentity/goaidentityservice.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/goaidentity/goaidentityservice.c b/src/goaidentity/goaidentityservice.c
index 8a804cd..14adf25 100644
--- a/src/goaidentity/goaidentityservice.c
+++ b/src/goaidentity/goaidentityservice.c
@@ -1664,6 +1664,8 @@ ensure_credentials_for_accounts (GoaIdentityService *self)
 
       ensure_account_credentials (self, object);
     }
+
+  g_list_free_full (accounts, g_object_unref);
 }
 
 static void
-- 
cgit v0.12

